elevation_layer/src/elevation_layer.cpp

```diff
@@ -0,0 +1,242 @@
+
+    void ElevationLayer::onInitialize() {
+        ros::NodeHandle nh("~/" + name_), g_nh;
+        rolling_window_ = layered_costmap_->isRolling();
+
+        ElevationLayer::matchSize();
+        current_ = true;
+        elevation_map_received_ = false;
+        filters_configuration_loaded_ = false;
+        global_frame_ = layered_costmap_->getGlobalFrameID();
+        elevation_layer_name_ = "elevation";
+        edges_layer_name_ = "edges";
+
+        // get our tf prefix
+        ros::NodeHandle prefix_nh;
+        const std::string tf_prefix = tf::getPrefixParam(prefix_nh);
+
+        // get parameters from config file
+        if(!nh.param("elevation_topic", elevation_topic_, std::string("")))
+        {
+            ROS_WARN("did not find elevation_topic, using default");
+        }
+        if(!nh.param("height_treshold", height_treshold_, 0.12))
+        {
+            ROS_WARN("did not find height_treshold, using default");
+        }
+        if( !nh.param("filter_chain_parameters_name", filter_chain_parameters_name_, std::string("elevation_filters")) )
+        {
+            ROS_WARN("did not find filter_chain_param_name, using default");
+        }
+        if(!nh.param("footprint_clearing_enabled", footprint_clearing_enabled_, true))
+        {
+            ROS_WARN("did not find footprint_clearing_enabled, using default");
+        }
+        if(!nh.param("combination_method", combination_method_, 2))
+        {
+            ROS_WARN("did not find combination_method, using default");
+        }
+        if(!nh.param("edges_sharpness_treshold", edges_sharpness_treshold_, 0.12))
+        {
+            ROS_WARN("did not find edges_sharpness_treshold, using default");
+        }
+        bool track_unknown_space;
+        nh.param("track_unknown_space", track_unknown_space, layered_costmap_->isTrackingUnknown());
+        if (track_unknown_space)
+            default_value_ = NO_INFORMATION;
+        else
+            default_value_ = FREE_SPACE;
+
+        // Subscribe to topic
+        elevation_subscriber_ = nh.subscribe(elevation_topic_, 1, &ElevationLayer::elevationMapCallback, this);
+        dsrv_ = NULL;
+        setupDynamicReconfigure(nh);
+
+        // Setup filter chain.
+        if (!filterChain_.configure(filter_chain_parameters_name_, nh)) {
+            ROS_WARN("Could not configure the filter chain!");
+        }else{
+            filters_configuration_loaded_ = true;
+        }
+    }
+
+    ElevationLayer::~ElevationLayer()
+    {
+        if (dsrv_)
+            delete dsrv_;
+    }
+
+    void ElevationLayer::updateBounds(double robot_x, double robot_y, double robot_yaw, double* min_x, double* min_y,
+                              double* max_x, double* max_y)
+    {
+        std::lock_guard<std::mutex> lock(elevation_map_mutex_);
+        if (rolling_window_)
+            updateOrigin(robot_x - getSizeInMetersX() / 2, robot_y - getSizeInMetersY() / 2);
+        if (!(enabled_ && elevation_map_received_))
```

Comment:
Makes more sense to do this before updating origin. Right?

Reply:
I think you still want your local costmap to move around with the robot even though the elevation_mapping is not received. But actually is kind of the same